wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java

```diff
@@ -581,10 +581,16 @@ public void onCommandFinish(Consumer<Command> action) {
    * directly or added to a composition.
    *
    * @param commands the commands to register
-   * @throws IllegalArgumentException if the given commands have already been composed.
+   * @throws IllegalArgumentException if the given commands have already been composed, or the array
+   *     of commands has duplicates.
    */
   public void registerComposedCommands(Command... commands) {
-    var commandSet = Set.of(commands);
+    Set<Command> commandSet;
+    try {
+      commandSet = Set.of(commands);
+    } catch (IllegalArgumentException e) {
```

Comment:
Is there any other reason this exception would be thrown? I'm a bit apprehensive that this would mask something else: maybe include the original exception as the `cause` parameter of the one you throw?

Reply:
The [docs for `Set.of(E...)`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Set.html#of(E...)) only note duplicate elements as a cause for throwing `IllegalArgumentException`, though if you're concerned about it I could just add some logic inside the catch block double-checking there's duplicates (and re-raise the exception if there aren't). (That way there wouldn't be another stacktrace cluttering the console and we can also be absolutely sure we identified the cause)

Reply:
No need to add logic.
If embedding the exception itself is too spammy (which I'm not sure about), then maybe embed only the original message.

Reply:
As it turns out, it's a moot point about whether another stacktrace would be too spammy- Apparently the error reporting in `RobotBase` will only display a single exception (see [RobotBase.java#L367](https://github.com/wpilibsuite/allwpilib/blob/main/wpilibj/src/main/java/edu/wpi/first/wpilibj/RobotBase.java#L367)), so I'll just include the `toString()` of the cause in the exception message.
I'll note that the `toString()` includes the exception class name, so the message can look like this: "java.lang.IllegalArgumentException: Cannot compose a command twice in the same composition! (Original exception: java.lang.IllegalArgumentException: duplicate element: edu.wpi.first.wpilibj2.command.RunCommand@776202e)". I'm leaving the class name in there for now since a subclass of `IllegalArgumentException` could be thrown, plus I think it provides context for the message, but I thought I'd send the error message in case anyone has thoughts.